src/frontend/state/libraryState.ts

```diff
@@ -0,0 +1,375 @@
+import {
+  FavouriteGame,
+  GameCollection,
+  GameInfo,
+  GameStatus,
+  HiddenGame
+} from 'common/types'
+import { Category, Platform } from 'frontend/types'
+import { makeAutoObservable } from 'mobx'
+import { getPlatformName } from 'frontend/helpers'
+import { DropdownItemType } from '@hyperplay/ui'
+import Fuse from 'fuse.js'
+import {
+  gogInstalledGamesStore,
+  gogLibraryStore,
+  libraryStore,
+  sideloadLibrary,
+  hyperPlayLibraryStore
+} from 'frontend/helpers/electronStores'
+import {
+  epicCategories,
+  gogCategories,
+  hyperPlayCategories,
+  sideloadedCategories
+} from 'frontend/helpers/library'
+
+class LibraryState {
+  epicLibrary: GameInfo[] = []
+  gogLibrary: GameInfo[] = []
+  sideloadedLibrary: GameInfo[] = []
+  hyperPlayLibrary: GameInfo[] = []
+  nonAvailbleGames: GameInfo[] = []
+  private gameStatuses: GameStatus[] = []
+
+  // cache list of games being installed
+  installing: string[] = []
+  libraryTopSection = ''
+  favouriteGames: GameCollection | undefined
+  hiddenGames: GameCollection | undefined
+  filterText = ''
+
+  // store
+  category: Category = 'all'
+
+  // filters
+  selectedFilter: DropdownItemType | undefined
```

Comment:
maybe we should define the filters as type? Something like:
```typescript
type Filter = 'asc' | 'desc'
const filterNames: Record<Filter, string> = {
    asc: 'Ascending'
    desc: 'Descending'
}
```

Reply:
We're already doing this for the category

Reply:
hmm. well the DropdownItemType is meant to be extensible to any dropdown item defined by the consuming code of the dropdown, so we can't type the id to be 
```
export type Filter = 'alphabeticalAscending' | 'alphabeticalDescending' | 'sortByInstalled'
```
in the ui library.

In the client I just added a `FilterItem` type that extends the dropdown item type and types the id prop to the filter values.

I think this addresses your concern. we only use the filter values in 3 or 4 lines in the code too so perfect typing isn't critical here imo